src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -443,19 +447,27 @@ pub async fn merge_imms_in_memory(
         });
     }
 
-    let mut versions: Vec<(EpochWithGap, HummockValue<Bytes>)> = Vec::new();
-
-    let mut pivot_last_delete_epoch = HummockEpoch::MAX;
-
-    for ((key, value), epoch) in items {
-        assert!(key >= pivot, "key should be in ascending order");
-        if key != pivot {
-            merged_payload.push((pivot, versions));
-            pivot = key;
-            pivot_last_delete_epoch = HummockEpoch::MAX;
-            versions = vec![];
-            let target_extended_user_key =
-                PointRange::from_user_key(UserKey::new(table_id, TableKey(pivot.as_ref())), false);
+    let mut full_key_tracker = FullKeyTracker::<Bytes>::new(FullKey::default());
```

Comment:
IIUC, here the The `full_key_tracker` is used as the original `pivot`. The `pivot` is initialized as `first_item_key`. Shall we also initialize the `full_key_tracker` as `first_item_key` with u64::MAX as epoch, and then we don't need an extra check on `!last_full_key.is_empty()`.